[Nicklelodeon] iP

Nicklelodeon opened the pull request:
### Duke

> “Your mind is for having ideas, not holding them.” – David Allen [Source](https://dansilvestre.com/productivity-quotes/)

**Duke frees your mind of having to remember things you need to do. It's,**

- text-based
- easy to learn
- SUPER FAST to use

**All you need to do is,**

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is FREE!

**Features:**

- [x]  Managing tasks
- [x]  Managing deadlines
- [ ]  Reminders (coming soon)

If you are Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        new Duke(FILE_PATH, DIRECTORY_PATH).run();
    }
}
```

Commit:
Add Gradle support

Nicklelodeon committed:
Level 1

Nicklelodeon committed:
Level 2

Nicklelodeon committed:
Level 3

Nicklelodeon committed:
Improved level 3

Nicklelodeon committed:
Level 4

Nicklelodeon committed:
TextUiTesting

Nicklelodeon committed:
Commit with throw

Nicklelodeon committed:
Level 5

Nicklelodeon committed:
Level 6

Nicklelodeon committed:
Add Enums and OOP

Nicklelodeon committed:
Edit for errors

Nicklelodeon committed:
Add level 7

Nicklelodeon committed:
Add Level 7 and improvements

Nicklelodeon committed:
Adds Level-8

Nicklelodeon committed:
Add A-MoreOOP with Ui class, Update duke.txt

Nicklelodeon committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Nicklelodeon committed:
Add A-Packages, A-Gradle, A-JUnit

Nicklelodeon committed:
Add JavaDoc to all non-private methods

Nicklelodeon committed:
Add CodingStandard updates to all classes

Nicklelodeon committed:
Add Level-9, which contains a FIND command that takes in a word and returns all tasks in the taskList that contains the word

Nicklelodeon committed:
Add A-Checkstyle, configured all classes to follow assigned checkstyle rules

Guo-KeCheng commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,125 @@
+                return false;
+            case EVENT:
+                try {
+                    String[] splitString = originalString.split("/");
+                    String description = splitString[0].replace("event", "");
+                    String startDate = splitString[1].replace("from", "");
+                    String endDate = splitString[2].replace("to", "");
```
Would it be better to abstract out the initial handling of string inputs from users?

Nicklelodeon replied:
Yep sounds good!

Guo-KeCheng commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,125 @@
+    Parser() { }
+
+    /**
+     * Reads input from user and splits it into cases based on Types specified
+     * in enum Types
+     * @param originalString Input from user
+     * @param taskList       List containing all current tasks
+     * @param ui             Ui that runs output
+     * @return Boolean based on input. Only returns true when command is bye,
+     *     which exits the chat bot. Else, returns false which continues the chat bot
+     * @throws NeroException Throws an exception depending on the exception faced
```
Love the formatting of java docs. Neat and easy to read.

Guo-KeCheng commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,80 @@
+                String line = sc.nextLine();
+                String[] cleanedInputs = line.split(" \\| ");
+                boolean isDone = (cleanedInputs[1].equals("0")) ? false : true;
+                if (cleanedInputs[0].equals("T")) {
```
Would be good to assign this to a variable to improve readability.

Nicklelodeon replied:
Sure, will add that in!

Guo-KeCheng commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,80 @@
+            String endDate) throws NeroException {
+        super(description);
+        try {
+            this.startDate = LocalDate.parse(startDate.trim());
+            this.endDate = LocalDate.parse(endDate.trim());
```
Good use of defensive code to handle potential errors.

Guo-KeCheng reviewed (commented):
LGTM. Just some nits to fix. Would also be good to group your files into sub-packages for better code organisation too.

Nicklelodeon committed:
Add level-10, which includes setting up GUI for Duke through javafx

Nicklelodeon committed:
Add new fat JAR file containing updated Duke, Update Duke to greet

MainWindow: When initialising, Duke does not print a standard greeting message

Printing a standard greeting message is a good addition and part of the ip requirements!

Nicklelodeon committed:
Add assertions

Event: Assertion checks that end date is after start date

Storage: Assertion checks that directory is properly created and present

Nicklelodeon committed:
Merge pull request #1 from Nicklelodeon/branch-A-Assertions

Add assertions

Nicklelodeon committed:
Add changes to improve code quality

Previously, logic used to clean code was done within the parseCommand() method, which might not be good coding practice, especially if the code cleaning was duplicated across different cases.
I have created a method cleanDeadline to do the string processing required for deadline tasks, and cleanEvent to do string processing required for event tasks.

Nicklelodeon committed:
Merge pull request #2 from Nicklelodeon/branch-A-CodeQuality

Add changes to improve code quality

Nicklelodeon committed:
Create new specific exceptions and exception package that groups all exceptions together. Add new Extension that allows rescheduling of tasks.

Previously, exceptions were not specific (only NeroException), which lead to duplicating of exception messages

Now, exceptions are more specific, which allows them to be used for specific situations. Removes duplication of exception messages.

New reschedule functionality aids users in keeping their task list flexible and adaptable to changes

Nicklelodeon committed:
Update README.md

Nicklelodeon committed:
Add new types of Exceptions, refactored code, repush BCD extension

Nicklelodeon committed:
Update README.md

Nicklelodeon committed:
Update README.rmd

Create User Guide

Nicklelodeon committed:
Update README.md

Nicklelodeon committed:
Add commands package

Previously code in parser to read inputs and execute commands was long and violated SLAP. Hence, I refactor code by creating different command classes and a command package.

Nicklelodeon committed:
Add comments, remove unnecessary methods

Nicklelodeon committed:
Update README.md

Nicklelodeon committed:
Add new exception to account for empty task list

Nicklelodeon committed:
Add Ui.png

Nicklelodeon committed:
Delete Ui.png

Nicklelodeon committed:
Add Ui.png

Nicklelodeon committed:
Delete Ui.png

Nicklelodeon committed:
Add Ui.png

Nicklelodeon committed:
Update README.md

Nicklelodeon committed:
Update README.md

Nicklelodeon committed:
Update User Guide
